api/v1alpha1/managedcertificate_types.go

```diff
@@ -0,0 +1,108 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make crd" to regenerate code after modifying this file
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ManagedCertificateSpec defines the desired state of ManagedCertificate.
+type ManagedCertificateSpec struct {
+	// Target defines the targets that the Certificate will be bound to.
+	// +kubebuilder:validation:Required
+	Target ManagedCertificateTarget `json:"target,omitempty"`
+
+	// DnsZone defines the DNS Zone that the ManagedCertificate will be applied to.
+	// +kubebuilder:validation:Required
+	DnsZone ManagedCertificateDnsZone `json:"dnsZone,omitempty"`
+
+	// DomainNames is a list of domain names that the Certificate will be issued for.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	DomainNames []string `json:"domainNames,omitempty"`
+
+	// ServiceAccount is the name of the ServiceAccount that will be used to connect to the Azure DNS Zone.
+	// +kubebuilder:validation:Required
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+}
+
+// ManagedCertificateTarget defines the targets that a Certificate will be bound to.
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:MaxProperties=1
+type ManagedCertificateTarget struct {
+	// Secret is the name of the Secret that will contain the Certificate.
+	Secret string `json:"secret,omitempty"`
```

Comment:
Is secret required?

Reply:
done. good catch

Reply:
actually wait no. min and max properties enforce this. I tricked myself

Reply:
I did that on purpose for future extensibility when we support just mounting to istio ingress pod directly